include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,200 @@
+		bool charge_fet_disabled;
+		/** BATTERY_CURRENT, */
+		int current;
+		/** BATTERY_CUT_OFF, */
+		bool cut_off;
+		/** BATTERY_CYCLE_COUNT, */
+		uint32_t cycle_count;
+		/** BATTERY_DISCONNECT_STATE, */
+		enum battery_disconnect_state disconnect_state;
+		/** BATTERY_FLAGS, */
+		uint32_t flags;
+		/** BATTERY_FULL_CHARGE_CAPACITY, */
+		uint32_t full_charge_capacity;
+		/** BATTERY_IMBALANCE_MV, */
+		int imbalance_mv;
+		/** BATTERY_PRESENT_STATE, */
+		enum battery_present battery_present_state;
+		/** BATTERY_REMAINING_CAPACITY, */
+		int remaining_capacity;
+		/** BATTERY_RUN_TIME_TO_EMPTY, */
+		uint32_t run_time_to_empty;
+		/** BATTERY_STATE_OF_CHARGE, */
+		uint8_t state_of_charge;
+		/** BATTERY_TEMPERATURE, */
+		int temperature;
+		/** BATTERY_VOLTAGE, */
+		int voltage;
+	};
+};
+
+/**
+ * Cut off charge from battery
+ *
+ * @return non-zero if the battery driver doesn't support cut off.
+ */
+typedef int (*battery_cut_off)(const struct device *dev);
+
+/**
+ * @brief Get a battery property
+ *
+ * @param dev Pointer to the battery device
+ * @param prop pointer to battery_get_property struct where the property struct
+ * field is set by the caller to determine what property is read from the
+ * battery device.
+ *
+ * @return 0 if successful, negative errno code if failure.
+ */
+typedef int (*battery_get_property_t)(const struct device *dev, struct battery_get_property *prop);
+
+/**
+ * Read time in seconds left when discharging.
+ *
+ * @param capacity	Destination for remaining time in seconds.
+ * @return non-zero if error.
+ */
+typedef int (*battery_time_to_empty)(const struct device *dev, int *seconds);
+
+/**
+ * Read time in seconds left to full capacity when charging.
+ *
+ * @param capacity	Destination for remaining time in seconds.
+ * @return non-zero if error.
+ */
+typedef int (*battery_time_to_full)(const struct device *dev, int *seconds);
+
+/**
+ * Calculate battery time in seconds, under an assumed current.
+ *
+ * @param rate		Current to use for calculation, in mA.
+ *			If > 0, calculates charging time; if < 0, calculates
+ *			discharging time; 0 is invalid and sets seconds=0.
+ * @param seconds	Destination for calculated time in seconds.
+ * @return non-zero if error.
+ */
+typedef int (*battery_time_at_rate)(const struct device *dev, int rate, int *seconds);
+
+/**
+ * Wait for battery stable.
+ *
+ * @return non-zero if error.
+ */
+typedef int (*battery_wait_for_stable)(const struct device *dev);
+
+/**
+ * Send the fuel gauge sleep command.
+ *
+ * @return	0 if successful, non-zero if error occurred
+ */
+typedef int (*battery_sleep_fuel_gauge)(const struct device *dev);
+
+__subsystem struct battery_driver_api {
+	battery_get_property_t get_property;
+
+	battery_time_at_rate time_at_rate;
+	battery_time_to_empty time_to_empty;
+	battery_time_to_full time_to_full;
+
+	battery_wait_for_stable wait_for_stable;
+	battery_sleep_fuel_gauge sleep_fuel_gauge;
+	battery_cut_off cut_off_charge;
```

Comment:
Please include APIs to support below features. These implementations are manufacture specific.
a.	Enable battery shipment mode
b.	Enable EMShut mode